# Import Libraries

In [2]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.graph_objs as go
import plotly.offline as py
py.init_notebook_mode(connected=True)
import plotly.tools as tls
import plotly.figure_factory as ff

from datetime import datetime

import warnings
warnings.filterwarnings("ignore")
print("All Libraries is Import")

All Libraries is Import


In [3]:
data=pd.read_csv('Data Sets/netflix_titles.csv')

In [4]:
data.head()

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA   1 Season   
3  September 24, 2021          2021  TV-MA   1 Season   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

In [5]:
data.shape

(8807, 12)

**Observation**
- In our datasets there is 8807 Rows and 12 Columns

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      8804 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


**Observation**
- there are 12 columns in dataset
- In our dataset only one columns is interger 

In [42]:
data.isnull().sum().sort_values(ascending=False)

director        2634
country          831
cast             825
date_added        10
rating             4
duration           3
show_id            0
type               0
title              0
release_year       0
listed_in          0
description        0
dtype: int64

**Observation**
- In our dataset 6 columns can contain null value
- The Director columns contains higher number of Nans value as compare to other

In [20]:
round(data.isnull().mean()*100,0)

show_id          0.0
type             0.0
title            0.0
director        30.0
cast             9.0
country          9.0
date_added       0.0
release_year     0.0
rating           0.0
duration         0.0
listed_in        0.0
description      0.0
dtype: float64

**Observation**
- In Director columns 30% contain Nans value
- In Columns cast and country contains 9% Nans value

In [22]:
data.dtypes

show_id         object
type            object
title           object
director        object
cast            object
country         object
date_added      object
release_year     int64
rating          object
duration        object
listed_in       object
description     object
dtype: object

# Data Cleaning

In [35]:
data_clean=data.copy()

In [36]:
data_clean.director.isnull().sum()

2634

In [37]:
fill_list=data_clean['director'].dropna().tolist()
data_clean['director']=data_clean['director'].fillna(pd.Series(np.random.choice(fill_list,size=len(data.index))))

In [40]:
data_clean.director.isnull().sum()

0

**Observation**
- After cleaning there is no Nans value in **Director** columns we randoms fill the value

In [43]:
data_clean.cast.isnull().sum()

825

In [44]:
fill_list=data_clean['cast'].dropna().tolist()
data_clean['cast']=data_clean['cast'].fillna(pd.Series(np.random.choice(fill_list, size=len(data.index))))

In [46]:
data_clean.cast.isnull().sum()

0

**Observation**
- After cleaning there is no Nans value in **Cast** columns we randoms fill the value

In [47]:
data_clean.country.isnull().sum()

831

In [48]:
fill_list = data_clean['country'].dropna().tolist()
data_clean['country'] = data_clean['country'].fillna(pd.Series(np.random.choice(fill_list , size = len(data.index))))

In [50]:
data_clean.country.isnull().sum()

0

**Observation**
- After cleaning there is no Nans value in **Country** columns we randoms fill the value

In [51]:
data_clean.date_added.isnull().sum()

10

In [52]:
data_clean['date_added']=data_clean['date_added'].fillna(method='ffill')

In [53]:
data_clean.date_added.isnull().sum()

0

**Observation**
- After cleaning there is no Nans value in **Data_added** columns we randoms fill the value

In [55]:
data_clean.rating.isnull().sum()

4

In [58]:
data_clean['rating'].replace(to_replace=np.nan, value="Not_rating",inplace=True)

In [59]:
data_clean.rating.isnull().sum()

0

**Observation**
- After cleaning there is no Nans value in **rating** columns we randoms fill the value

In [62]:
data_clean['duration'].isnull().sum()


3